community/kernel/src/main/java/org/neo4j/kernel/impl/store/CommonAbstractStore.java

```diff
@@ -510,12 +510,12 @@ protected void checkStoreOk()
     /**
      * Returns the next id for this store's {@link IdGenerator}.
      *
-     * @return The next free id
+     * @return The next free id, or -1 if not initialized
      */
     @Override
     public long nextId()
     {
-        return idGenerator.nextId();
+        return idGenerator != null ? idGenerator.nextId() : -1;
```

Comment:
Callers don't really check for -1, do they? I think this should throw exception instead because it is an exceptional condition.